src/map/atcommand.cpp

```diff
@@ -7799,9 +7799,12 @@ ACMD_FUNC(showmobs)
 		return 0;
 	}
 
-	if(mob_id == strtol(mob_name, nullptr, 10) && !mob->jname.empty())
-		strcpy(mob_name, mob->jname.c_str());    // --ja--
-		//strcpy(mob_name, mob->name.c_str());    // --en--
+	if (mob_id == strtol(mob_name, nullptr, 10)) {
+		if (battle_config.mob_names_type == 0 && !mob->name.empty())
+			strcpy(mob_name, mob->name.c_str());    // --en--
+		else if (battle_config.mob_names_type == 1 && !mob->jname.empty())
+			strcpy(mob_name, mob->jname.c_str());    // --ja--
```

Comment:
Here you should probably just obey the config.
If they did not set the respective name it is a user fault.
Maybe we could validate in loadingFinished of mob db based on the config?

Additionally this logic could be hidden inside a function of s_mob_db.
```cpp
std::string& s_mob_db::get_name(){
	if( battle_config.mob_names_type == 0 ){
		return this->name;
	}else{
		return this->jname;
	}
}
```

This function could be used anywhere, where you already have s_mob_db looked up. :)